src/core/mac/mac.cpp

```diff
@@ -1269,6 +1269,14 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
 
             mBroadcastTransmitCount = 0;
         }
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+        // Verify Enh-ACK integrity by checking its MIC
+        if ((aError == OT_ERROR_NONE) && (aAckFrame != nullptr))
+        {
+            aError = (ProcessEnhAckSecurity(aFrame, *aAckFrame) == OT_ERROR_NONE) ? OT_ERROR_NONE : OT_ERROR_NO_ACK;
```

Comment:
Just have a question here: if the ack is not authenticated successfully(maybe there are attackers), shall we directly jump to the end or let upper layer handle this with a error code? I think there should be the right ACK soon. If we handle this TX here, then this TX has finished. If we got the right ACK later, we still got failure.